pkg/reconciler/workload/resource/resource_controller.go

```diff
@@ -279,7 +285,23 @@ func (c *Controller) processResource(ctx context.Context, key string) error {
 	if err != nil {
 		return err
 	}
-	obj, exists, err := inf.Informer().GetIndexer().GetByKey(key)
+
+	lclusterName, namespace, name, err := kcpcache.SplitMetaClusterNamespaceKey(key)
+	if err != nil {
+		logger.Error(err, "failed to split key, dropping")
+		return nil
+	}
+	// TODO(skuznets): can we figure out how to not leak this detail up to this code?
+	// I guess once the indexer is using kcpcache.MetaClusterNamespaceKeyFunc, we can just use that formatter ...
+	var indexKey string
+	if namespace != "" {
+		indexKey += namespace + "/"
+	}
+	if !lclusterName.Empty() {
+		indexKey += lclusterName.String() + "|"
```

Comment:
This is the right format? namespace/cluster|name?

Reply:
That's what our fork of k8s is doing right now, yes ...